### ISSUE TITLE
[idea/feature] Be compatible with RISC-V architecture

### ISSUE THREAD

**Issue by sbechet** [redox-os/redox#1141]
- [x] I agree that I have searched opened and closed issues to prevent duplicates.
## Description

Port redox on RISC-V because it is fun.

See [RISCVEMU](https://bellard.org/riscvemu/) and [128-bits rust support](https://github.com/rust-lang/rfcs/pull/1504).

**Comment by batonius** [comment]
The privileged ISA hasn't been stabilized yet tho: https://riscv.org/specifications/privileged-isa/ .

**Comment by Ddnirvana** [comment]
Good idea! Even linux has supported RISC-V architecture. BTW, spike and qemu are also great simulator choices for porting.

**Comment by BearzRobotics** [comment]
I know its a bit expensive, but I wanted to post that they now have a computer capable of running Linux.
https://www.crowdsupply.com/sifive/hifive-unleashed

**Comment by boomshroom** [comment]
The biggest issue is the fact the riscv is still an experimental LLVM target and not complete. It might be possible to compile the codebase, but it would require manually building Rust so it enables RISC-V for LLVM, or to target a prebuilt LLVM that supports it.

**Comment by toxik-io** [comment]
I was curious about doing this ages ago. But i think it was llvm that was the problem.
